```
[ERROR] Tests run: 17, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 75.795 s <<< FAILURE! - in org.apache.hadoop.hdfs.server.datanode.TestBPOfferService
[ERROR] testMissBlocksWhenReregister(org.apache.hadoop.hdfs.server.datanode.TestBPOfferService)  Time elapsed: 15.474 s  <<< FAILURE!
java.lang.AssertionError
    at org.junit.Assert.fail(Assert.java:87)
    at org.junit.Assert.assertTrue(Assert.java:42)
    at org.junit.Assert.assertTrue(Assert.java:53)
    at org.apache.hadoop.hdfs.server.datanode.TestBPOfferService.testMissBlocksWhenReregister(TestBPOfferService.java:350)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

This is a flaky test:
- "TestBPOfferService#testMissBlocksWhenReregister fails intermittently" 
- https://issues.apache.org/jira/browse/HDFS-15646

--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminDefaultMonitor.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminDefaultMonitor.java
@@ -275,7 +275,6 @@ public class DatanodeAdminDefaultMonitor extends DatanodeAdminMonitorBase
             + "{}.", dn, e);
         getPendingNodes().add(dn);
         toRemove.add(dn);
-        unhealthyDns.remove(dn);
       } finally {
         iterkey = dn;
       }

--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDecommission.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDecommission.java
@@ -1724,7 +1724,7 @@ public class TestDecommission extends AdminStatesBaseTest {
     GenericTestUtils.waitFor(() -> decomManager.getNumTrackedNodes() == numDeadNodes
             && decomManager.getNumPendingNodes() == numLiveDecommNodes
             && liveDecommNodes.stream().allMatch(
-        node -> node.getAdminState().equals(AdminStates.DECOMMISSION_INPROGRESS)),
+                node -> node.getAdminState().equals(AdminStates.DECOMMISSION_INPROGRESS)),
         500, 30000);
     assertThat(liveDecommNodes)
         .as("Check all live decommissioning nodes queued in DatanodeAdminManager")
@@ -1803,7 +1803,7 @@ public class TestDecommission extends AdminStatesBaseTest {
       }
       return decomManager.getNumTrackedNodes() == 0 && decomManager.getNumPendingNodes() == 0
           && deadNodeProps.keySet().stream().allMatch(
-          node -> node.getAdminState().equals(AdminStates.DECOMMISSIONED));
+              node -> node.getAdminState().equals(AdminStates.DECOMMISSIONED));
     }, 500, 30000);
   }
 

